src/errors.rs

```diff
@@ -26,15 +26,15 @@ pub enum Error {
     InvalidState,
 }
 
-impl From<Utf8Error> for Error {
+impl<SerialError> From<Utf8Error> for Error<SerialError> {
```

Comment:
I personally prefer single-letter names for generic arguments, because you can directly see whether `Error<SerialError>` is a concrete type, or a parametrized type. Something like this:

```rust
impl<SerialError> From<Error<SerialError>> for JoinError<SerialError> {
```

...looks very confusing to me. This seems more clear:

```rust
impl<S> From<Error<S>> for JoinError<S> {
```

Could you change it?